File: src/libqpropertyprivate.zig
const C = @import("qt6c");

/// https://doc.qt.io/qt-6/quntypedpropertydata.html
pub const quntypedpropertydata = struct {
    /// New constructs a new QUntypedPropertyData object.
    ///
    /// ``` other: ?*C.QUntypedPropertyData ```
    pub fn New(other: ?*anyopaque) ?*C.QUntypedPropertyData {
        return C.QUntypedPropertyData_new(@ptrCast(other));
    }

    /// New2 constructs a new QUntypedPropertyData object and invalidates the source QUntypedPropertyData object.
    ///
    /// ``` other: ?*C.QUntypedPropertyData ```
    pub fn New2(other: ?*anyopaque) ?*C.QUntypedPropertyData {
        return C.QUntypedPropertyData_new2(@ptrCast(other));
    }

    /// CopyAssign shallow copies `other` into `self`.
    ///
    /// ``` self: ?*QUntypedPropertyData, other: ?*QUntypedPropertyData ```
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        C.QUntypedPropertyData_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    ///
    /// ``` self: ?*QUntypedPropertyData, other: ?*QUntypedPropertyData ```
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        C.QUntypedPropertyData_MoveAssign(@ptrCast(self), @ptrCast(other));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: ?*C.QUntypedPropertyData ```
    pub fn QDelete(self: ?*anyopaque) void {
        C.QUntypedPropertyData_Delete(@ptrCast(self));
    }
};

File: src/libqpropertyprivate.cpp
#include <QPropertyProxyBindingData>
#include <QUntypedPropertyData>
#include <qpropertyprivate.h>
#include "libqpropertyprivate.h"
#include "libqpropertyprivate.hxx"

QUntypedPropertyData* QUntypedPropertyData_new(const QUntypedPropertyData* other) {
    return new QUntypedPropertyData(*other);
}

QUntypedPropertyData* QUntypedPropertyData_new2(QUntypedPropertyData* other) {
    return new QUntypedPropertyData(std::move(*other));
}

void QUntypedPropertyData_CopyAssign(QUntypedPropertyData* self, QUntypedPropertyData* other) {
    *self = *other;
}

void QUntypedPropertyData_MoveAssign(QUntypedPropertyData* self, QUntypedPropertyData* other) {
    *self = std::move(*other);
}

void QUntypedPropertyData_Delete(QUntypedPropertyData* self) {
    delete self;
}
